packages/plugin-ext/src/plugin/tasks/tasks.ts

```diff
@@ -227,33 +271,31 @@ export class TasksExtImpl implements TasksExt {
         }
     }
 
+    private toTaskExecution(execution: TaskExecutionDto): theia.TaskExecution {
+        const result = {
+            task: converter.toTask(execution.task),
+            terminate: () => {
+                this.proxy.$terminateTask(execution.id);
+            }
+        };
+        if (execution.task.executionId) {
```

Comment:
I see that you use `executionId` field to check if it's `CustomExecution` task.
So - if only tasks with `CustomExecution` have this field - should we provide some clear/more readable way for that...

Reply:
What do you have in mind? I'm not trying to do any typing here.

Reply:
> only tasks with CustomExecution have this field
is it correct?

Reply:
As far as I know every task of **any** type (not only `customExecution`) can have `execution` - the object is created when a task is started ([$onDidStartTask()](https://github.com/eclipse-theia/theia/blob/f71bfec8649fef13e3319d37bcf28487a769b291/packages/plugin-ext/src/plugin/tasks/tasks.ts#L88) => [getTaskExecution()](https://github.com/eclipse-theia/theia/blob/f71bfec8649fef13e3319d37bcf28487a769b291/packages/plugin-ext/src/plugin/tasks/tasks.ts#L238-L242)) and [is stored to the map with executions](https://github.com/eclipse-theia/theia/blob/f71bfec8649fef13e3319d37bcf28487a769b291/packages/plugin-ext/src/plugin/tasks/tasks.ts#L244) using `executionId`.

I see that: 
- within one of your previous PRs `executionId` [field](https://github.com/eclipse-theia/theia/blob/f71bfec8649fef13e3319d37bcf28487a769b291/packages/plugin-ext/src/common/plugin-api-rpc.ts#L1367) was added to the `TaskDto` interface
- within current PR you are using the field as a **marker** of a task with `customExecution`
first, you create the field, like: 
````
if (dto && CustomExecution.is(task.execution!)) {
    dto.executionId = this.addProvidedCustomExecution(task.execution);
````
and in the line 281 above you check - if the field exists - then `result.task.execution = this.getCustomExecution(execution.task.executionId);`) 

So, in my comment above I meant that for me it's not obvious that `executionId` [field](https://github.com/eclipse-theia/theia/blob/f71bfec8649fef13e3319d37bcf28487a769b291/packages/plugin-ext/src/common/plugin-api-rpc.ts#L1367) of the `TaskDto` interface is a marker for `customExecution` tasks.

Reply:
> for me it's not obvious that executionId field of the TaskDto interface is a marker for customExecution tasks.

If you think the executionId field is named confusingly, I'm all up for renaming it to something more clear. But since it wasn't introduced in this PR, it's not a concern for this PR.